docs/user/development.rst

```diff
@@ -2,151 +2,233 @@ Development
 
+To send us general feedback, questions or ideas for improvement, please post on
+`our mailing list`_.
 
-Code coverage can be tested with
+If you have a very concrete feature proposal, add it to the `issue tracker on
+GitHub`_:
 
-.. code:: bash
+* Explain how it would work, and link to a scientific paper if applicable.
+* Keep the scope as narrow as possible, to make it easier to implement.
 
-    $ py.test
 
-Testing will take over 5 minutes for the first run, but less than a minute for
-subsequent runs when Theano can reuse compiled code.
+Report bugs
+~~~~~~~~~~~
 
-You can add a so called pre-commit-hook to git to run all tests automatically
-before you commit. The hook is installed by copying the following code to
-`.git/hooks/pre-commit` and making this file executable:
+Report bugs at the `issue tracker on GitHub`_.
+If you are reporting a bug, please include:
 
-.. code:: bash
+* your Lasagne and Theano version.
+* steps to reproduce the bug, ideally reduced to a few Python commands.
+* the results you obtain, and the results you expected instead.
 
-    #!/usr/bin/env bash
+If you are unsure whether the behavior you experience is a bug, or if you are
+unsure whether it is related to Lasagne or Theano, please just ask on `our
+mailing list`_ first.
 
-    # Stash uncommited changes to make sure the commited ones will work
-    git stash -q --keep-index
-    # Run tests
-    py.test
-    RETURN_CODE=$?
-    # Pop stashed changes
-    git stash pop -q
 
-    exit $RETURN_CODE
+Fix bugs
+~~~~~~~~
 
+Look through the GitHub issues for bug reports. Anything tagged with "bug" is
+open to whoever wants to implement it. If you discover a bug in Lasagne you can
+fix yourself, by all means feel free to just implement a fix and not report it
+first.
 
-Documentation
--------------
 
-The documentation is generated with `Sphinx <http://sphinx-doc.org/latest/index.html>`_.
+Implement features
+~~~~~~~~~~~~~~~~~~
 
-Sphinx makes use of `reStructured Text <http://openalea.gforge.inria.fr/doc/openalea/doc/_build/html/source/sphinx/rest_syntax.html>`_
+Look through the GitHub issues for feature proposals. Anything tagged with
+"feature" or "enhancement" is open to whoever wants to implement it. If you
+have a feature in mind you want to implement yourself, please note that Lasagne
+has a fairly narrow focus and we strictly follow a set of :ref:`design
+principles <lasagne-philosopy>`, so we cannot guarantee upfront that your code
+will be included. Please do not hesitate to just propose your idea in a GitHub
+issue or on the mailing list first, so we can discuss it and/or guide you
+through the implementation.
 
-You should also `install pylearn2 <http://deeplearning.net/software/pylearn2/#download-and-installation>`_
-to prevent warnings while generating the documentation.
 
-The documentation can be built with ``make html``.
+Write documentation
+~~~~~~~~~~~~~~~~~~~
 
-The documentation is written in numpydoc syntax. Information about numpydoc
```

Comment:
We're linking to this one instead: https://github.com/numpy/numpy/blob/master/doc/HOWTO_DOCUMENT.rst.txt
Should be fine as that's the one you need to obey, shouldn't it?